cfn-guard-rulegen/src/lib.rs

```diff
@@ -44,7 +44,10 @@ fn gen_rules(cfn_resources: HashMap<String, Value>) -> Vec<String> {
     for (name, cfn_resource) in cfn_resources {
         trace!("{} is {:?}", name, &cfn_resource);
         let props: HashMap<String, Value> =
-            serde_json::from_value(cfn_resource["Properties"].clone()).unwrap();
+            match serde_json::from_value(cfn_resource["Properties"].clone()) {
+                Ok(s) => s,
+                Err(_) => continue
+            };
```

Comment:
Nice! Working well on resources with no explicit properties now:
```bash
$ bin/cfn-guard-rulegen --version
CloudFormation Guard RuleGen 0.5.0
$ bin/cfn-guard-rulegen template.yaml # resources without explicit properties
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: null, expected a map", line: 0, column: 0)', src/lib.rs:47:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
$ cat template.yaml
Resources:
  Resource:
    Type: AWS::SNS::Topic # resource without explicit properties
$ gh pr checkout 28 # brew install github/gh/gh
$ make cfn-guard-rulegen
$ bin/cfn-guard-rulegen template.yaml
```

@nathanataws `Ok()` line is a no-op, right? Assuming `Ok()` just has to be handled in order to continue on `Err()`?

Reply:
`unwrap()` is a way of "cheating" the type system to say "I'm positive this [Result](https://doc.rust-lang.org/std/result/enum.Result.html) will come back without an Err.  That's "safe" if you know, for instance, that your Result has already been checked somewhere else and this is just, say, getting the captures back from the same regex on the same string, just in a different place.  Anywhere else that your assumptions aren't guaranteed should really be using the `match` operator to force evaluation for both the "success" (OK) and "failure" (Err) modes.  

Match is a pattern matcher that allows you to unwrap the "s" in `OK(s)` for more computation or `Err` (and the err payload if you care about it).  In this case, if there *is* a Properties section of the template, we get back an `OK` and that SerDe Value is in the `s` which then gets bound to `props` has a `HashMap<String, Value>` type.  If there isn't a properties section, you get `Err` and guard just continues the loop by grabbing the next resource and trying it all over again. etc, etc

Reply:
My mistake in `rulegen` was assuming that no one would every try to generate rules for a template that had a resource with no properties in it.  That was obviously incorrect and I need to not be lazy about and just handle the both cases of the Result properly.  As part of this release, I also went back through the guard code in other places and eliminated any other unwrap() that was based on assumptions I couldn't support.  The practical result _should_ be that there might be a handled failure if some garbage is passed in, but we shouldn't be getting panics like the one you saw in rulegen.

Reply:
Thanks again for the bug report.  This was a good catch and a good reminder to myself.